pandas/core/internals/blocks.py

```diff
@@ -1783,6 +1782,23 @@ def get_values(self, dtype=None):
     def to_dense(self):
         return np.asarray(self.values)
 
+    def to_native_types(self, slicer=None, na_rep="nan", quoting=None, **kwargs):
+        """override to use ExtensionArray astype for the conversion"""
+        values = self.values
+        if slicer is not None:
+            values = values[slicer]
+        mask = isna(values)
+
+        try:
+            values = values.astype(str)
+            values[mask] = na_rep
+        except Exception:
+            # eg SparseArray does not support setitem, needs to be converted to ndarray
+            return super().to_native_types(slicer, na_rep, quoting, **kwargs)
+
+        # we are expected to return a 2-d ndarray
```

Comment:
In theory I can do all of this in the base class method, putting the above code behind a `if self.is_extension`. But it's kind of the point of subclasses to override those things, instead of keeping the different implementations of all subclasses in the parent class with if statements? 

But I don't care much, happy to move it as well. Some other Blocks (like DatetimeBlock) also override this method, so overriding it here as well seems consistent with that.

Reply:
no biggie